include/picongpu/fields/currentDeposition/Esirkepov/bitPacking.hpp

```diff
@@ -0,0 +1,90 @@
+/* Copyright 2013-2022 Axel Huebl, Heiko Burau, Rene Widera, Sergei Bastrakov
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/fields/currentDeposition/Esirkepov/Line.hpp"
+
+#include <cstdint>
+
+
+namespace picongpu
+{
+    namespace currentSolver
+    {
+        /** Helper base class for Esirkepov to manage bit packing operations to encode the status of the start and end
+         * particle to optimize the algorithm.
+         * Bit-packing for the status is required to keep the register footprint low.
+         */
+        namespace bitpacking
+        {
+            enum class Status
+            {
+                /** Marke that a particle trajectory is crossing the assignment cell border.
+                 *
+                 * For performance reason it is required that LEAVE_CELL is represented by the first bit. This allow
+                 * that we can use getValue() to querry the integer value. It is used to optimize the loop contition in
+                 * the Esirkepov implemenation.
+                 */
```

Comment:
I think this doxygen comment should be moved a bit up, to be in front of `enum class`